eth-providers/src/base-provider.ts

```diff
@@ -546,7 +546,7 @@ export abstract class BaseProvider extends AbstractProvider {
    */
   estimateGas = async (transaction: Deferrable<TransactionRequest>): Promise<BigNumber> => {
     const resources = await this.estimateResources(transaction);
-    return resources.gas;
+    return resources.gas.mul(30); // TODO: this is a temp gas limit, we should optimize gas limit later
```

Comment:
estimateResources provides the storage limit as well, so should use `gas_limit + storage_entry_limit * storage_entry_deposit / tx_fee_per_gas` to return the value

Reply:
done